"use strict";(self.webpackChunkdocs=self.webpackChunkdocs||[]).push([[88574],{3905:(e,t,n)=>{n.d(t,{Zo:()=>p,kt:()=>u});var i=n(67294);function r(e,t,n){return t in e?Object.defineProperty(e,t,{value:n,enumerable:!0,configurable:!0,writable:!0}):e[t]=n,e}function o(e,t){var n=Object.keys(e);if(Object.getOwnPropertySymbols){var i=Object.getOwnPropertySymbols(e);t&&(i=i.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),n.push.apply(n,i)}return n}function a(e){for(var t=1;t<arguments.length;t++){var n=null!=arguments[t]?arguments[t]:{};t%2?o(Object(n),!0).forEach((function(t){r(e,t,n[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(n)):o(Object(n)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(n,t))}))}return e}function s(e,t){if(null==e)return{};var n,i,r=function(e,t){if(null==e)return{};var n,i,r={},o=Object.keys(e);for(i=0;i<o.length;i++)n=o[i],t.indexOf(n)>=0||(r[n]=e[n]);return r}(e,t);if(Object.getOwnPropertySymbols){var o=Object.getOwnPropertySymbols(e);for(i=0;i<o.length;i++)n=o[i],t.indexOf(n)>=0||Object.prototype.propertyIsEnumerable.call(e,n)&&(r[n]=e[n])}return r}var l=i.createContext({}),c=function(e){var t=i.useContext(l),n=t;return e&&(n="function"==typeof e?e(t):a(a({},t),e)),n},p=function(e){var t=c(e.components);return i.createElement(l.Provider,{value:t},e.children)},d={inlineCode:"code",wrapper:function(e){var t=e.children;return i.createElement(i.Fragment,{},t)}},m=i.forwardRef((function(e,t){var n=e.components,r=e.mdxType,o=e.originalType,l=e.parentName,p=s(e,["components","mdxType","originalType","parentName"]),m=c(n),u=r,h=m["".concat(l,".").concat(u)]||m[u]||d[u]||o;return n?i.createElement(h,a(a({ref:t},p),{},{components:n})):i.createElement(h,a({ref:t},p))}));function u(e,t){var n=arguments,r=t&&t.mdxType;if("string"==typeof e||r){var o=n.length,a=new Array(o);a[0]=m;var s={};for(var l in t)hasOwnProperty.call(t,l)&&(s[l]=t[l]);s.originalType=e,s.mdxType="string"==typeof e?e:r,a[1]=s;for(var c=2;c<o;c++)a[c]=n[c];return i.createElement.apply(null,a)}return i.createElement.apply(null,n)}m.displayName="MDXCreateElement"},44151:(e,t,n)=>{n.r(t),n.d(t,{assets:()=>l,contentTitle:()=>a,default:()=>d,frontMatter:()=>o,metadata:()=>s,toc:()=>c});var i=n(87462),r=(n(67294),n(3905));const o={id:"bioinformatics",title:"Bioinformatics",sidebar_label:"Bioinformatics"},a=void 0,s={unversionedId:"use-cases/bioinformatics",id:"use-cases/bioinformatics",title:"Bioinformatics",description:"Introduction",source:"@site/mage/use-cases/bioinformatics.md",sourceDirName:"use-cases",slug:"/use-cases/bioinformatics",permalink:"/docs/mage/use-cases/bioinformatics",draft:!1,editUrl:"https://github.com/memgraph/docs/tree/master/mage/use-cases/bioinformatics.md",tags:[],version:"current",frontMatter:{id:"bioinformatics",title:"Bioinformatics",sidebar_label:"Bioinformatics"},sidebar:"mage",previous:{title:"Temporal Graph Networks",permalink:"/docs/mage/algorithms/machine-learning-graph-analytics/temporal-graph-networks"},next:{title:"Computer Security",permalink:"/docs/mage/use-cases/computer-security"}},l={},c=[{value:"Introduction",id:"introduction",level:2},{value:"Applying graphs",id:"applying-graphs",level:2},{value:"Protein-protein interaction networks",id:"protein-protein-interaction-networks",level:3},{value:"Sequence similarity networks",id:"sequence-similarity-networks",level:3},{value:"Disease networks",id:"disease-networks",level:3},{value:"Where to next?",id:"where-to-next",level:2}],p={toc:c};function d(e){let{components:t,...o}=e;return(0,r.kt)("wrapper",(0,i.Z)({},p,o,{components:t,mdxType:"MDXLayout"}),(0,r.kt)("h2",{id:"introduction"},"Introduction"),(0,r.kt)("p",null,(0,r.kt)("strong",{parentName:"p"},"Bioinformatics")," is a subdiscipline of biology and computer science that\nfocuses on discovering, storing, analyzing, and disseminating biological data\nsuch as DNA and amino acid sequences. This kind of data often comes in the form\nof highly complex and interconnected networks. While most of the data is\ncurrently stored in relational databases, the problem of complicated domain\nmodels and evergrowing data diversity presents a challenge to data scientists. "),(0,r.kt)("p",null,"Memgraph as a graph database can efficiently map the relationships in such\nhighly interconnected networks in the field of biology. From molecule to protein\ninteractions, almost everything can be modeled using graphs and explored with\ngraph algorithms."),(0,r.kt)("p",null,(0,r.kt)("img",{alt:"memgraph-graph-algorithm-applications-bioinformatics",src:n(35325).Z,width:"1000",height:"300"})),(0,r.kt)("h2",{id:"applying-graphs"},"Applying graphs"),(0,r.kt)("h3",{id:"protein-protein-interaction-networks"},"Protein-protein interaction networks"),(0,r.kt)("p",null,"Proteins rarely act individually, they are often in interaction with other\nproteins to perform some function. Protein-protein interaction networks (PPIN)\nare used to simplify and model these complex interactions. In PPIN, proteins are\nrepresented as nodes. Interactions between proteins are described by edges\nconnecting the corresponding nodes."),(0,r.kt)("p",null,"Researches have discovered an important property of PPINs and consequentially\nenabled us to develop methods for identifying essential proteins. Most proteins\nin PPINs interact with only several other proteins, meaning that they do not\nhave many connections. However, there is a small number of proteins that\ninteract with almost every other protein. These proteins, included in many\ninteractions, ",(0,r.kt)("a",{parentName:"p",href:"https://www.ebi.ac.uk/training/online/courses/network-analysis-of-protein-interaction-data-an-introduction/building-and-analysing-ppins/topological-ppin-analysis/centrality-analysis/"},"can be\ndetected"),"\nin the network as highly connected nodes, called hubs and they are very likely\nessential proteins."),(0,r.kt)("h3",{id:"sequence-similarity-networks"},"Sequence similarity networks"),(0,r.kt)("p",null,"A ",(0,r.kt)("a",{parentName:"p",href:"http://sfld.rbvi.ucsf.edu/archive/django/web/networks/index.html"},"sequence similarity\nnetwork")," (SNN)\nenables the visualization of relationships among protein sequences. The proteins\nwhich are related in some way are most often grouped in clusters. The graph is\nmade up of nodes that represent proteins, while the edges indicate similarity in\namino acid sequence."),(0,r.kt)("p",null,"SNNs are used to explore relationships in large and diverse sets of sequences\nbecause the computational resources needed for traditional methods of analysis\n(for example, phylogenetic trees) would be unfeasible, due to the difficulty in\ngenerating accurate multiple alignments."),(0,r.kt)("h3",{id:"disease-networks"},"Disease networks"),(0,r.kt)("p",null,"Systems biology experiments create an enormous amount of data of multiple\nmodalities. Because of its complexity and rich semantics, this kind of data\npresents a hurdle for standard storage and analysis solutions."),(0,r.kt)("p",null,"Graph databases provide a flexible solution for the integration of multiple\ntypes of biological data and facilitate exploratory data mining to support\nhypothesis generation."),(0,r.kt)("h2",{id:"where-to-next"},"Where to next?"),(0,r.kt)("p",null,"This text is a ",(0,r.kt)("strong",{parentName:"p"},"summary")," of one area that fits perfectly with the application\nof graphs. Therefore, we would like to have you with us when ",(0,r.kt)("strong",{parentName:"p"},"implementing"),"\nsome of these solutions. Share ",(0,r.kt)("strong",{parentName:"p"},"opinions"),", ",(0,r.kt)("strong",{parentName:"p"},"experiences")," and ",(0,r.kt)("strong",{parentName:"p"},"problems"),"\nyou encounter when working with ",(0,r.kt)("strong",{parentName:"p"},"Memgraph")," on our ",(0,r.kt)("a",{parentName:"p",href:"https://discord.gg/memgraph"},"Discord\nserver"),". We are here for you and we will help you\nalong the way."))}d.isMDXComponent=!0},35325:(e,t,n)=>{n.d(t,{Z:()=>i});const i=n.p+"assets/images/memgraph-graph-algorithm-applications-bioinformatics-6d34a3e6a05e36cc0c6dd99e6d88bb18.png"}}]);